salt/modules/win_shortcut.py

```diff
@@ -0,0 +1,471 @@
+                icon_location, icon_index = shortcut.IconLocation.split(",")
+                icon_location = salt.utils.path.expand(icon_location)
+            if shortcut.WindowStyle:
+                window_style = WINDOW_STYLE[shortcut.WindowStyle]
+            if shortcut.WorkingDirectory:
+                working_dir = salt.utils.path.expand(shortcut.WorkingDirectory)
+        else:
+            target = shortcut.TargetPath
+
+        return {
+            "arguments": arguments,
+            "description": description,
+            "hot_key": hot_key,
+            "icon_index": int(icon_index),
+            "icon_location": icon_location,
+            "path": path,
+            "target": target,
+            "window_style": window_style,
+            "working_dir": working_dir,
+        }
+
+
+def _set_info(
+    path,
+    target,
+    arguments="",
+    description="",
+    hot_key="",
+    icon_index=0,
+    icon_location="",
+    window_style="Normal",
+    working_dir="",
+):
+    r"""
+    The main worker function for creating and modifying shortcuts. the `create`
+    and `modify` functions are wrappers around this function.
+
+    Args:
+
+        path (str): The full path to the shortcut
+
+        target (str): The full path to the target
+
+        arguments (str, optional): Any arguments to be passed to the target
+
+        description (str, optional): The description for the shortcut. This is
+            shown in the ``Comment`` field of the dialog box. Default is an
+            empty string
+
+        hot_key (str, optional): A combination of hot Keys to trigger this
+            shortcut. This is something like ``Ctrl+Alt+D``. This is shown in
+            the ``Shortcut key`` field in the dialog box. Default is an empty
+            string. Available options are:
+
+            - Ctrl
+            - Alt
+            - Shift
+            - Ext
+
+        icon_index (int, optional): The index for the icon to use in files that
+            contain multiple icons. Default is 0
+
+        icon_location (str, optional): The full path to a file containing icons.
+            This is shown in the ``Change Icon`` dialog box by clicking the
+            ``Change Icon`` button. If no file is specified and a binary is
+            passed as the target, Windows will attempt to get the icon from the
+            binary file. Default is an empty string
+
+        window_style (str, optional): The window style the program should start
+            in. This is shown in the ``Run`` field of the dialog box. Default is
+            ``Normal``. Valid options are:
+
+            - Normal
+            - Minimized
+            - Maximized
+
+        working_dir (str, optional): The full path to the working directory for
+            the program to run in. This is shown in the ``Start in`` field of
+            the dialog box.
+
+    Returns:
+        bool: True if successful
+    """
+    path = salt.utils.path.expand(path)
```

Comment:
Can we verify this is a real path ?

Reply:
Done

Reply:
Actually, the `salt.utils.path.expand` function returns a real path